src/ffi/datetime.rs

```diff
@@ -201,13 +252,21 @@ impl Deref for PyDateTimeAPI {
 /// such as if you do not want the first call to a datetime function to be
 /// slightly slower than subsequent calls.
 pub unsafe fn PyDateTime_IMPORT() -> &'static PyDateTime_CAPI {
-    // PyDateTime_CAPSULE_NAME is a macro in C
-    let PyDateTime_CAPSULE_NAME = CString::new("datetime.datetime_CAPI").unwrap();
+    // PyPy excepts the C-API to be initialized via PyDateTime_Import, so trying to use
```

Comment:
s/excepts/expects

Also, I think this comment is *slightly* confusing, I would reword as:

```rust
// PyPy uses a different mechanism to initialize the datetime C API
// and so must call PyDateTime_Import
```

I can give perhaps better suggestion for the wording if you can explain what `PyDateTime_Import` is? Is that a PyPy-specific function that imports the datetime capsule?

Reply:
So from what I understand cpython expects users to import the datetime C-API via `PyDateTime_IMPORT`, which is a macro that imports the capsule.

PyPy only tries to emulate the API, so it exposes the name of the macro as a seperate symbol, which it expects to be called.
So importing the capsule directly (which has no effect in cpython since this is a pure macro) does not work. 

cpython:
![image](https://user-images.githubusercontent.com/2467993/54947819-e0892c00-4f43-11e9-874a-8b72733a0350.png)

helpful comment:
![image](https://user-images.githubusercontent.com/2467993/54947938-2fcf5c80-4f44-11e9-8b5f-b9a902171d00.png)

If you have better suggestions for wording to make this clearer I'll be happy to head 😄

Reply:
But the capsule *is* the C API. The macros aren't even available outside of C (which is why we have to define them here ourselves).

Does PyPy not have a mechanism to emulate capsule imports as well?

Reply:
It does export the symbol `PyCapsule_Import`, but when I've tried it the module was not initialized.

It seems they are doing some extra work when "PyDateTime_IMPORT" is called via an _PyDateTime_IMPORT internal symbol.

(source for the internal initialization code)
![image](https://user-images.githubusercontent.com/2467993/54950617-542e3780-4f4a-11e9-9d9a-5fef5ec10d04.png)

Also, this is the test testing cpyext support for datetime, notice how what is defined as a macro in cpython, is considered a whole different beast here (when linking against their version of Python.h)
![image](https://user-images.githubusercontent.com/2467993/54950690-7cb63180-4f4a-11e9-9dbb-69cf46feabdf.png)

Reply:
Let me look into this a bit more closely. My understanding is that even though `datetime` is one of the only standard library modules to actually use them, capsules are the preferred mechanism for handling C API imports.

This particular conditional compilation doesn't need to be a blocker or anything, but it might be worth trying to support capsule imports upstream in PyPy, if only because it makes `cpyext` a better compatibility layer for C extensions.

Reply:
awesome, let me know if there's anything you find out about this! 

This may be a specific case with `datetime` they implemented specific behavior for, but I guess if they intend to support other c-extensions they would need to make it behave the same.